src/StatsAPI.jl

```diff
@@ -1,3 +1,21 @@
 module StatsAPI
 
+#    pairwise(f, x[, y])
+#
+# Return a matrix holding the result of applying `f` to all possible pairs
```

Comment:
are we sure it will always be a matrix? Maybe it should be a more general `n`-dimensional pairwise (but admittedly then it should probably be tuple-wise)

Reply:
In particular as `x` and `y` are allowed to be any iterators what should happen with a 0-dimensional iterator (e.g. when `x` is a number)

Reply:
> are we sure it will always be a matrix? Maybe it should be a more general `n`-dimensional pairwise (but admittedly then it should probably be tuple-wise)

"pairwise" seems to imply there are only two elements, right? IIRC we mentioned the possibility of having a more general `outer` function, but I find `pairwise` nicer to compute pairwise correlations, and it's already used by Distances.

> In particular as `x` and `y` are allowed to be any iterators what should happen with a 0-dimensional iterator (e.g. when `x` is a number)

Well the number of dimensions of the iterator doesn't matter, only the number of elements it contains. So a number gives a 1×1 matrix.

Reply:
OK